File: TransCoder_tokenized_test_set_functions/python/SUBARRAY_NO_PAIR_SUM_DIVISIBLE_K.py
def subarrayDivisibleByK ( arr, n, k ) :
    mp = [ 0 ] * 1000
    s = 0;
    e = 0;
    maxs = 0;
    maxe = 0;
    
    mp [ arr [ 0 ] % k ] = mp [ arr [ 0 ] % k ] + 1;
    
    for i in range ( 1, n ) :
        mod = arr [ i ] % k
        while ( mp [ k - mod ] != 0 or ( mod == 0 and mp [ mod ] != 0 ) ) :
            mp [ arr [ s ] % k ] = mp [ arr [ s ] % k ] - 1
            s = s + 1
        mp [ mod ] = mp [ mod ] + 1
        e = e + 1
        if ( ( e - s ) > ( maxe - maxs ) ) :
            maxe = e
            maxs = s
    print ( "The maximum size is {} and the " . format ( ( maxe - maxs + 1 ) ) ) for i in range ( maxs, maxe + 1 ) :
        print ( "{} " . format ( arr [ i ] ), end = "" )
    

File: TransCoder_tokenized_test_set_functions/cpp/SUBARRAY_NO_PAIR_SUM_DIVISIBLE_K.cpp
void subarrayDivisibleByK ( int arr [ ], int n, int k ) {
  map < int, int > mp;
  int s = 0, e = 0, maxs = 0, maxe = 0;
  mp [ arr [ 0 ] % k ] ++;
  for ( int i = 1;
  i < n;
  i ++ ) {
    int mod = arr [ i ] % k;
    while ( mp [ k - mod ] != 0 || ( mod == 0 && mp [ mod ] != 0 ) ) {
      mp [ arr [ s ] % k ] --;
      s ++;
    }
    mp [ mod ] ++;
    e ++;
    if ( ( e - s ) > ( maxe - maxs ) ) {
      maxe = e;
      maxs = s;
    }
  }
  cout << "The maximum size is " << maxe - maxs + 1 << " and " "the subarray is as follows\n";
  for ( int i = maxs;
  i <= maxe;
  i ++ ) cout << arr [ i ] << " ";
}